config/data_scrubber.go

```diff
@@ -17,11 +19,19 @@ var (
 		"secret", "credentials", "stripetoken"}
 )
 
+const (
+	defaultCacheTTL = 100
+)
+
 // DataScrubber allows the agent to blacklist cmdline arguments that match
 // a list of predefined and custom words
 type DataScrubber struct {
 	Enabled           bool
 	SensitivePatterns []*regexp.Regexp
+	seenProcess       map[string]struct{}
+	cachedCmdlines    map[string][]string
```

Comment:
You don't need `seenProcess` since `cachedCmdlines` has the same key structure.

Reply:
I kept it now since I store just the scrubbed cmdlines in this map